ls1intum/Artemis
# `Communication`: Show author's role as icon in course on post title

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Currently, many communication tools like Zulip allow setting authorities to users. This allows others to easily understand the author's role and assess their trustworthiness. Having this feature would be beneficial in Artemis.

### Description
We add an icon in front of the author name in all Postings with a corresponding tooltip. The icon reveals the user's highest authority for that course and provides information about that user's role within its' tooltip. You are very welcome to suggest changes to these explanations and especially their German translations. :)

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Tutor
- 1 Editor
- 1 Course with Postings enabled

1. Log in to Artemis
2. Navigate to Course Communication
3. Make sure that the corresponding icon with the appropriate tooltip is displayed before the author name for each Posting. Editors and Tutors are assigned the "Tutor" icon.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| PostService.java | 81% | 98% |
| posting-header.directive.ts | 92.3% | 100% |

### Screenshots


![image](https://user-images.githubusercontent.com/67344158/181028985-6e779cf1-4368-4e8e-8a65-fee2c68cded2.png)
https://user-images.githubusercontent.com/67344158/181029286-56e8d9d3-4e85-40ec-b16e-7a39c55fec47.mp4
